### ISSUE TITLE
Got Error: DeveloperError Exception of type 'Google.GoogleSignIn+SignInException' was thrown.

### ISSUE THREAD

**Issue by ChiefBreakeverything** [googlesamples/google-signin-unity#49]
I've run into this error while using the example scene as directed. Unlike outcomes other users have reported before, I'm 99% certain the issue isn't related to the spelling of the webClientID string (its copied straight from the google-services credentials json file, which itself is fresh from the credentials dashboard). I've checked over lots of the aspects that I imagine could be responsible, here's a list of the things I've triple-checked:

- Created an API key and OAuth 2.0 client ID on Google Cloud Platform and enabled the required APIs.
- webClientID string in the SignInSampleScript scene; it uses the client_ID where the client_type is '3' from the google-services.json file, which was pulled directly from the 'API & Settings' -> 'Credentials' page on Google Cloud Platform.
- The package_name from the same file where client_type is '1'. This is copied and pasted into Unity's player settings.
- The keystore file, key alias and their passwords are present in Unity's player settings. We aren't using debug.keystore and have created our own keystore file.
- The SHA-1 fingerprint has been copied using the [keytool -exportcert -list -v -alias <KeystoreAliasNameWasHere> -keystore <PathToKeystoreFileWasHere>] console command successfully and pasted it into the OAuth2.0 part of the 'API & Settings' -> 'Credentials' page on Google Cloud Platform.
- I've run AndroidResolver from the Unity -> Assets -> Play Services Resolver menu after each change, just to ensure up-to-dated-ness.

I managed to log each of the GoogleSignInStatusCode enum flag states returned with the GoogleSignIn.SignInException task and the results look like this:

- Error.Status.Flag[Success] Is it this? [True]
- Error.Status.Flag[ApiNotConnected] Is it this? [False]
- Error.Status.Flag[Canceled] Is it this? [True]
- Error.Status.Flag[Interrupted] Is it this? [False]
- Error.Status.Flag[InvalidAccount] Is it this? [True]
- Error.Status.Flag[Timeout] Is it this? [False]
- Error.Status.Flag[DeveloperError] Is it this? [True]
- Error.Status.Flag[InternalError] Is it this? [False]
- Error.Status.Flag[NetworkError] Is it this? [False]
- Error.Status.Flag[Error] Is it this? [False]

I'm not sure in what to make of these results (it succeeded and was cancelled?) but it's interesting to note that it returned InvalidAccount. I'm just using my standard Google account which I use for all development purposes, perhaps there's some sort of disconnect between it and this project?

I'm not sure what exactly to check or try next, any help would be greatly appreciated.

**Comment by Syrlander** [comment]
Creating the OAuth client under the Google Play Console has solved the problem for my group as well, therefor I would assume that's the problem most other developers have when seeing the SigninException and not a problem with the package. Huge thanks @ChiefBreakeverything.

**Comment by kevinfoley** [comment]
@ChiefBreakeverything Perhaps you should close the issue if you found a solution? Although this solution doesn't make sense. The API Access section of Google Play is for setting up software that can edit and publish apps on Google Play using your Google Play Developer Account.

**Comment by karam89** [comment]
After days of losing efforts making it work, I just had to update the OAuth Consent screen (From Google Developers Console),

After updating the info on that screen, the exception has gone, and the sign in worked well

I've not read this information anywhere else

**Comment by ghost** [comment]
Well finally solved it after 2 days of struggle!

Solution: Do make sure in the "OAuth consent screen" in google api console you've added name of your application properly.

**Comment by ninjarlz** [comment]
@karam89  yup, exactly, one has to make any change in that section, so as to make it possible to save new settings of 'OAuth Consent screen'. Rly strange, but works!

**Comment by Mukikaizoku** [comment]
@ChiefBreakeverything Thanks for this post---I inferred from your post that `debug.keystore` doesn't work for some reason and that you have to use your app's release keystore---doing that was what got the DeveloperError fixed for me.

During the process, I did add the Google Play Console Oauth client, but I removed it add things are still working--as @kevinfoley mentioned, it's unclear why that would be needed as it's function seems unrelated.

**Comment by iliaJelly** [comment]
SOLVED !!!
I had this problem for 2 days
I solved it by switching over to WIFI, mobile data blocks google account management on my Android.

This also makes me think if Google SignIn is a good tool since its not very foolproof on Wifi-less environments like public transportation and outdoors.

**Comment by BlissGDev** [comment]
It's happening the same to me and I'm not using Firebase.

I have the setup as follows:
-Android, iOS and Web keys configured.
-Added the correct SHA-1 string into the Android Key
-Added to the Unity Project the Web Id and pasted the configuration from Play Console
-Build with release keys
-Added accounts to testers on the Cloud Console Platform

But I keep getting the same error:
`DeveloperError Exception of type `'Google.GoogleSignIn+SignInException'`

Any suggestion on what I might be missing ?
